### PR TITLE
User will not enter empty software stack

### PR DESCRIPTION
**Pull request by ace2107**

### CODE REVIEW

File: thoth_user_api/api_v1.py [hunk 1]
```diff
@@ -61,20 +70,22 @@ def analyze(image: str, analyzer: str, debug: bool=False, timeout: int=None, cpu
 def solve(solver: str, packages: dict, debug: bool=False,
           cpu_request: str=None, memory_request: str=None, transitive: bool=False):
     """Run a solver in a restricted namespace."""
+    if check_empty(packages) == True:
+        return 0
```
Comment (fridex) [inline thoth_user_api/api_v1.py]: what about something like:

```python
packages = packages.pop('requirements', '' ) # see line below
if not packages:
   # error handling here
```

File: thoth_user_api/api_v1.py [hunk 2]
```diff
@@ -61,20 +70,22 @@ def analyze(image: str, analyzer: str, debug: bool=False, timeout: int=None, cpu
...
     packages = packages.pop('requirements', '')
     params = locals()
     try:
         return {
-            'pod_id': run_solver(**params),
-            'parameters': params
+        'pod_id': run_solver(**params),
+        'parameters': params
```
Comment (fridex) [inline thoth_user_api/api_v1.py]: Please check indentation here and below as well.

**Comment by fridex** [comment]
BTW take a look at schema we request in request:

https://github.com/thoth-station/user-api/blob/master/thoth_user_api/swagger.yaml#L494-L504

You can enforce requirements to be non-empty.

**Comment by goern** [comment]
@ace2107 ping

**Comment by goern** [comment]
ping @ace2107

**Comment by fridex** [comment]
@ace2107 take a look at https://spacetelescope.github.io/understanding-json-schema/reference/string.html#length as an example for this. The YAML file schema definition is linked above.

**Comment by goern** [comment]
@ace2107 please fix that PR today, its idling around for about 20 days.

File: thoth_user_api/swagger.yaml [hunk 3]
```diff
@@ -499,7 +499,7 @@ definitions:
     properties:
       requirements:
         type: string
-        minlength: 2
+        minLength: 2
```
Comment (fridex) [inline thoth_user_api/swagger.yaml]: This is too much, there are packages that have name that consists of only one letter - see [1]. Anyway, the code you did before can be deleted based on this schema configuration.

[1] https://pypi.org/project/q/

**Comment by fridex** [comment]
Closing in respect to #132

**Pull request closed without merging**